log/logger.go

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2020 - for information on the respective copyright owner
+// see the NOTICE file and/or the repository at
+// https://github.com/hyperledger-labs/perun-node
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var logger *logrus.Logger = nil
+
+// Logger is a for now, a type alias of Logrus.FieldLogger that defines a broad interface for logging.
+type Logger = logrus.FieldLogger
+
+// InitLogger sets the internal logger instance to the given level and log file.
+// This function should be called exactly once and subsequent calls return an error.
+// Logs to stdout if logFile is an empty string.
+func InitLogger(levelStr, logFile string) error {
+	if logger != nil {
+		return errors.New("logger already initialized")
+	}
+
+	newLogger := logrus.New()
+	level, err := logrus.ParseLevel(levelStr)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	newLogger.SetLevel(level)
+	if logFile == "" {
+		newLogger.SetOutput(os.Stdout)
+	} else {
+		f, err := os.OpenFile(filepath.Clean(logFile), os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0o600)
+		if err != nil {
+			return errors.WithStack(err)
+		}
+		newLogger.SetOutput(f)
```

Comment:
newLogger.SetOutput(f) is not covered in unit test.

Reply:
Add a test to cover this in https://github.com/hyperledger-labs/perun-node/pull/71/commits/0a45d9fcb47dde34e28664c735cf8a27cbf980d0